app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/TransportNotificationsProvider.kt

```diff
@@ -0,0 +1,52 @@
+package de.tum.`in`.tumcampusapp.component.ui.transportation
+
+import android.content.Context
+import android.support.v4.app.NotificationCompat
+import de.tum.`in`.tumcampusapp.R
+import de.tum.`in`.tumcampusapp.component.notifications.model.AppNotification
+import de.tum.`in`.tumcampusapp.component.notifications.NotificationsProvider
+import de.tum.`in`.tumcampusapp.component.ui.transportation.model.efa.Departure
+import de.tum.`in`.tumcampusapp.component.ui.transportation.model.efa.StationResult
+import de.tum.`in`.tumcampusapp.utils.Const
+
+class TransportNotificationsProvider(
+        context: Context,
+        private val departures: List<Departure>,
+        private val station: StationResult) : NotificationsProvider(context) {
+
+    override fun getNotificationBuilder(): NotificationCompat.Builder {
+        return NotificationCompat.Builder(context, Const.NOTIFICATION_CHANNEL_MVV)
+                .setSmallIcon(R.drawable.ic_notification)
+                .setColor(notificationColorAccent)
+    }
+
+    override fun getNotifications(): List<AppNotification> {
+        return emptyList()
+
+        // TODO: Add intelligent scheduling (last lecture)
```

Comment:
This is still TODO, please implement that before finishing the PR

Reply:
I don't know if we can actually figure out the location of the building, I think we only have the room number.
To figure out if one would use public transport we would probably need to query MVV with an coordinate request about the nearest station for the two lectures.

I would make it a little bit simpler for now and schedule only a single alarm half an hour before the last lecture of the day ends (i.e. next lecture is not today).
For now I'd also unconditionally notify the user. We can always improve the solution later :tada:

Reply:
> Once the alarm goes off, check if the user is at the location and if so, load and show the departures

This is overkill. Simplifying it as Philipp said sound reasonable.